src/golang/cmd/server/handler/connect_integration.go

```diff
@@ -248,3 +277,117 @@ func validateAirflowConfig(
 
 	return http.StatusOK, nil
 }
+
+// checkIntegrationSetStorage returns whether this integration should be used as the storage layer.
+func checkIntegrationSetStorage(ctx context.Context, svc integration.Service, conf auth.Config) (bool, error) {
+	if svc != integration.S3 {
+		// Only S3 integrations can be used for the storage layer
+		return false, nil
+	}
+
+	data, err := conf.Marshal()
+	if err != nil {
+		return false, err
+	}
+
+	var c integration.S3Config
+	if err := json.Unmarshal(data, &c); err != nil {
+		return false, err
+	}
+
+	return c.UseAsStorage, nil
+}
+
+func setIntegrationAsStorage(conf auth.Config) error {
+	data, err := conf.Marshal()
+	if err != nil {
+		return err
+	}
+
+	var c integration.S3Config
+	if err := json.Unmarshal(data, &c); err != nil {
+		return err
+	}
+
+	storageConfig, err := convertS3IntegrationtoStorageConfig(&c)
+	if err != nil {
+		return err
+	}
+
+	// Change global storage config
+	return config.UpdateStorage(storageConfig)
+}
+
+func convertS3IntegrationtoStorageConfig(c *integration.S3Config) (*shared.StorageConfig, error) {
+	// Users provide AWS credentials for an S3 integration via one of the following:
+	//  1. AWS Access Key and Secret Key
+	//  2. Credentials file content
+	//  3. Credentials filepath and profile name
+	// The S3 Storage implementation expects the AWS credentials to be specified via a
+	// filepath and profile name, so we must convert the above to the correct format.
+	storageConfig := &shared.StorageConfig{
+		Type: shared.S3StorageType,
+		S3Config: &shared.S3Config{
+			Bucket: c.Bucket,
+		},
+	}
+	switch c.Type {
+	case integration.AccessKeyS3ConfigType:
+		// AWS access and secret keys need to be written to a credentials file
+		path := filepath.Join(config.AqueductPath(), "storage", uuid.NewString())
+		f, err := os.Create(path)
+		if err != nil {
+			return nil, err
+		}
+		defer f.Close()
+
+		credentialsContent := fmt.Sprintf(
+			"[default]\naws_access_key_id=%s\naws_secret_access_key=%s\n",
+			c.AccessKeyId,
+			c.SecretAccessKey,
+		)
+		if _, err := f.WriteString(credentialsContent); err != nil {
+			return nil, err
+		}
+
+		storageConfig.S3Config.CredentialsPath = path
+		storageConfig.S3Config.CredentialsProfile = "default"
+	case integration.ConfigFileContentS3ConfigType:
+		// The credentials content needs to be written to a credentials file
+		path := filepath.Join(config.AqueductPath(), "storage", uuid.NewString())
+		f, err := os.Create(path)
+		if err != nil {
+			return nil, err
+		}
+		defer f.Close()
+
+		// Determine profile name by looking for [profile_name]
+		i := strings.Index(c.ConfigFileContent, "[")
+		if i < 0 {
+			return nil, errors.New("Unable to determine AWS credentials profile name.")
+		}
+
+		j := strings.Index(c.ConfigFileContent, "]")
+		if j < 0 {
+			return nil, errors.New("Unable to determine AWS credentials profile name.")
+		}
```

Comment:
Might be better to use a regex library? https://gobyexample.com/regular-expressions

Just curious, do we control the format of `c.ConfigFileContent`? Eg. can we always expect a "[<profile_name>]" format or is it user-controlled

Reply:
I'm not 100% sure, I think in most cases we can. Wei would have more context, but I don't want to ping him on PTO. @cw75 Do you have any thoughts here?

I'm thinking that if the file does not have a profile name we can just go with `default`?

Reply:
Actually after thinking about this more, since this is an AWS credentials file convention, I think we should throw an error if the file is not formatted correctly. Because, even if we assume `default`, the actual AWS API calls will fail when the AWS client tries to find the `default` profile in the config file.

Reply:
Yeah I think its safest just to throw.